clientconn.go

```diff
@@ -411,6 +419,7 @@ func (cc *ClientConn) Close() error {
 	}
 	cc.state = Shutdown
 	cc.stateCV.Broadcast()
+	cc.events.Finish()
```

Comment:
suggest using "defer" for this at line 416 because the exit path at line 418 also needs to call Finish.